sodium/src/kdf.rs

```diff
@@ -0,0 +1,104 @@
+//! This module provides access to libsodium
+
+use super::{check_init, secbuf::SecBuf};
+use crate::{
+    error::{SodiumError, SodiumResult},
+    util::check_buf_len,
+};
+
+pub const CONTEXTBYTES: usize = rust_sodium_sys::crypto_kdf_CONTEXTBYTES as usize;
+
+/// Derive a subkey from a parent key
+/// ****
+/// @param {SecBuf} out - Empty Buffer to be used as output
+///
+/// @param {number} index - subkey index
+///
+/// @param {Buffer} context - eight bytes context
+///
+/// @param {SecBuf} parent - the parent key to derive from
+
+pub fn derive(
+    out: &mut SecBuf,
+    index: u64,
+    context: &mut SecBuf,
+    parent: &mut SecBuf,
+) -> SodiumResult<()> {
+    check_init();
+    {
+        let out = out.read_lock();
+        let o = out.len();
+        let context = context.read_lock();
+        let c = context.len();
+        if check_buf_len(o) {
```

Comment:
the output size doesn't need to be aligned (unless you are using a secure secbuf)... but it does have to be > `crypto_kdf_BYTES_MIN` and < `crypto_kdf_BYTES_MAX`